agenta-web/src/components/TestSetTable/TestsetTable.tsx

```diff
@@ -194,12 +189,124 @@ const TestsetTable: React.FC<testsetTableProps> = ({mode}) => {
         })
 
         setColumnDefs(newColumnDefs)
+
         setRowData(newRowData)
         if (gridRef.current) {
             gridRef.current.setColumnDefs(newColumnDefs)
         }
     }
 
+    const HeaderComponent = (params: any) => {
+        const {eGridHeader} = params
+        const [scopedInputValues, setScopedInputValues] = useState(
```

Comment:
Any specific reason for maintaining a duplicate state of inputValues here? I think we can avoid this.

Reply:
so trying to change the inputValues state from within the header component, breaks the modal state to edit a column name once the onChange event for the input is triggered.

Reply:
Yeah, so we can just use the `displayName` state and update the displayName on input change rather than updating the inputValues. And only in handleSave we append the displayName field in the inputValues array. That we its more simpler and we can avoid the duplicated setInputValues state